server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -789,8 +801,77 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
       for (Entry<Key,Value> entry : scanner) {
         Key key = entry.getKey();
         Value value = entry.getValue();
+
+        final KeyExtent keyExtent = KeyExtent.fromMetaRow(key.getRow());
+
+        // Keep track of the last Key Extent seen so we can use it to fence
+        // of RFiles when merging the metadata
+        if (lastExtent != null && !keyExtent.equals(lastExtent)) {
+          previousKeyExtent = lastExtent;
+        }
+
+        // Special case for now to handle the highest/stop tablet which is where files are
+        // merged to so we need to handle the deletes on update here as it won't be handled later
+        // TODO: Can this be re-written to not have a special edge case and make it simpler?
+        if (keyExtent.equals(stopExtent)) {
+          if (previousKeyExtent != null
```

Comment:
Good question, I just put the null check in there but you are right it would only happen in the single tablet case. You are right that i think the merge should just do nothing if it detects it's one tablet